platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/DtoConverter.java

```diff
@@ -158,170 +126,109 @@ public static ItemReference toItemReference(FolderEntry folder,
-        List<AccessControlEntry> acl = null;
-        try {
-            acl = project.getPermissions();
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
-        if (acl != null) {
-            final List<String> permissions = new LinkedList<>();
-            if (acl.isEmpty()) {
-                // there is no any restriction at all
-                permissions.add("all");
-            } else {
-                for (AccessControlEntry accessControlEntry : acl) {
-                    final Principal principal = accessControlEntry.getPrincipal();
-                    if ((Principal.Type.USER == principal.getType() && currentUser.getId().equals(principal.getName()))
-                        || (Principal.Type.USER == principal.getType() && "any".equals(principal.getName()))
-                        || (Principal.Type.GROUP == principal.getType() && currentUser.isMemberOf(principal.getName()))) {
-
-                        permissions.addAll(accessControlEntry.getPermissions());
-                    }
-                }
-            }
-            dto.withPermissions(permissions);
-        }
-
-        try {
-            dto.withCreationDate(project.getCreationDate());
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
-
-        try {
-            dto.withModificationDate(project.getModificationDate());
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
-
-        try {
-            dto.withVisibility(project.getVisibility());
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
+    /**
+     * The method tries to provide as much as possible information about project.If get error then save information about error
+     * with 'problems' field in ProjectConfigDto.
+     *
+     * @param project
+     *         project from which we need get information
+     * @param serviceUriBuilder
+     *         service for building URI
+     * @return an instance of {@link ProjectConfigDto}
+     * @throws InvalidValueException
+     */
+    public static ProjectConfigDto toProjectConfig(Project project,
+                                                   UriBuilder serviceUriBuilder) throws ForbiddenException,
+                                                                                        ServerException,
+                                                                                        ProjectTypeConstraintException,
+                                                                                        ValueStorageException {
+        String name = project.getName();
+        String path = project.getPath();
+
+        ProjectConfigDto projectConfigDto = DtoFactory.getInstance().createDto(ProjectConfigDto.class);
```

Comment:
Just a hint: In this case `DtoFactory.newDto(ProjectConfigDto.class)` looks much better